lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -103,23 +106,28 @@ public long getImmutableTimestamp() {
     }
 
     @Override
-    public LockRefreshTokenV2Adapter lock(LockRequestV2 request) {
+    public Optional<LockTokenV2> lock(LockRequestV2 request) {
         LockRequest legacyRequest = toLegacyLockRequest(request);
 
         LockRefreshToken legacyToken = lockAnonymous(legacyRequest);
-        return new LockRefreshTokenV2Adapter(legacyToken);
+        return Optional.ofNullable(legacyToken).map(LockRefreshTokenV2Adapter::new);
     }
 
     @Override
-    public void waitForLocks(WaitForLocksRequest request) {
+    public boolean waitForLocks(WaitForLocksRequest request) {
         LockRequest legacyRequest = toLegacyWaitForLocksRequest(request.getLockDescriptors());
 
         lockAnonymous(legacyRequest);
+        return true;
```

Comment:
It looks like legacyRequest always has an infinite timeout, from `toLegacyWaitForLocksRequest`. So if we assume the earlier statement was correct (this will be optional.empty() iff we timed out), which I think is the case, then lockAnonymous always returns nonnull.